src/engine/value.rs

```diff
@@ -261,6 +261,20 @@ where
     }
 }
 
+impl TryFrom<Value> for HashMap<String, Value> {
+    type Error = Error;
+
+    fn try_from(value: Value) -> Result<HashMap<String, Value>, Error> {
+        match value {
+            Value::Map(hm) => Ok(hm),
+            _ => Err(Error::TypeConversionFailed {
+                src: "Value".to_string(),
```

Comment:
I would recommend putting a formatted debug output of the `value` here, as opposed to just the Value type name, as it will provide richer information for debugging.

Reply:
Fixed.